Failed to build with rustc 1.23.0-nightly (33374fa9d 2017-11-20)

Due to: rust-lang/rust#45225

Log:
```
error[E0599]: no method named `layout` found for type `&'tcx rustc::ty::TyS<'tcx>` in the current scope
    --> .cargo/registry/src/github.com-1ecc6299db9ec823/clippy_lints-0.0.172/src/utils/mod.rs:1024:8
     |
1024 |     ty.layout(cx.tcx, cx.param_env)
     |        ^^^^^^

error: aborting due to previous error

error: Could not compile `clippy_lints`.
warning: build failed, waiting for other jobs to finish...
error: failed to compile `clippy v0.0.172`, intermediate artifacts can be found at `/tmp/cargo-install.q3IwltO2ONEC`

Caused by:
  build failed
```

--- a/clippy_lints/src/utils/mod.rs
+++ b/clippy_lints/src/utils/mod.rs
@@ -9,6 +9,7 @@ use rustc::lint::{LateContext, Level, Lint, LintContext};
 use rustc::session::Session;
 use rustc::traits;
 use rustc::ty::{self, Ty, TyCtxt};
+use rustc::ty::layout::LayoutOf;
 use rustc_errors;
 use std::borrow::Cow;
 use std::env;
@@ -1021,9 +1022,9 @@ pub fn is_try(expr: &Expr) -> Option<&Expr> {
 }
 
 pub fn type_size<'a, 'tcx>(cx: &LateContext<'a, 'tcx>, ty: Ty<'tcx>) -> Option<u64> {
-    ty.layout(cx.tcx, cx.param_env)
+    (cx.tcx, cx.param_env).layout_of(ty)
         .ok()
-        .map(|layout| layout.size(cx.tcx).bytes())
+        .map(|layout| layout.size.bytes())
 }
 
 /// Returns true if the lint is allowed in the current context

